File: clients/java/src/main/java/komus24/model/Order.java
package komus24.model;

import komus24.util.StreamUtil;

/**
 * Player's orders
 */
public class Order {
    /**
     * TODO - Document
     */
    private komus24.model.VehicleOrder[] vehicles;

    /**
     * TODO - Document
     */
    public komus24.model.VehicleOrder[] getVehicles() {
        return vehicles;
    }

    /**
     * TODO - Document
     */
    public void setVehicles(komus24.model.VehicleOrder[] value) {
        this.vehicles = value;
    }

    public Order(komus24.model.VehicleOrder[] vehicles) {
        this.vehicles = vehicles;
    }

    /**
     * Read Order from input stream
     */
    public static Order readFrom(java.io.InputStream stream) throws java.io.IOException {
        komus24.model.VehicleOrder[] vehicles;
        vehicles = new komus24.model.VehicleOrder[StreamUtil.readInt(stream)];
        for (int vehiclesIndex = 0; vehiclesIndex < vehicles.length; vehiclesIndex++) {
            komus24.model.VehicleOrder vehiclesElement;
            vehiclesElement = komus24.model.VehicleOrder.readFrom(stream);
            vehicles[vehiclesIndex] = vehiclesElement;
        }
        return new Order(vehicles);
    }

    /**
     * Write Order to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, vehicles.length);
        for (komus24.model.VehicleOrder vehiclesElement : vehicles) {
            vehiclesElement.writeTo(stream);
        }
    }

    /**
     * Get string representation of Order
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("Order { ");
        stringBuilder.append("vehicles: ");
        stringBuilder.append("[ ");
        for (int vehiclesIndex = 0; vehiclesIndex < vehicles.length; vehiclesIndex++) {
            if (vehiclesIndex != 0) {
                stringBuilder.append(", ");
            }
            komus24.model.VehicleOrder vehiclesElement = vehicles[vehiclesIndex];
            stringBuilder.append(String.valueOf(vehiclesElement));
        }
        stringBuilder.append(" ]");
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/kotlin/src/main/kotlin/komus24/model/Order.kt
package komus24.model

import komus24.util.StreamUtil

/**
 * Player's orders
 */
class Order {
    /**
     * TODO - Document
     */
    var vehicles: Array<komus24.model.VehicleOrder>

    constructor(vehicles: Array<komus24.model.VehicleOrder>) {
        this.vehicles = vehicles
    }

    /**
     * Write Order to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, vehicles.size)
        for (vehiclesElement in vehicles) {
            vehiclesElement.writeTo(stream)
        }
    }

    /**
     * Get string representation of Order
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Order { ")
        stringBuilder.append("vehicles: ")
        stringBuilder.append("[ ")
        var vehiclesIndex = 0
        for (vehiclesElement in vehicles) {
            if (vehiclesIndex != 0) {
                stringBuilder.append(", ")
            }
            stringBuilder.append(vehiclesElement)
            vehiclesIndex++
        }
        stringBuilder.append(" ]")
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Order from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Order {
            var vehicles: Array<komus24.model.VehicleOrder>
            vehicles = Array(StreamUtil.readInt(stream), {
                var vehiclesElement: komus24.model.VehicleOrder
                vehiclesElement = komus24.model.VehicleOrder.readFrom(stream)
                vehiclesElement
            })
            return Order(vehicles)
        }
    }
}